Unable to retrieve status for gRPC service

### Describe the bug

When running IntegrationTests with `@grpcService` and not an HTTP server, Quarkus is not able to retrieve the status of the running process and it fails with the following error:

```
2021-11-08 14:26:58,647 INFO  [io.qua.grp.run.GrpcServerRecorder] (vert.x-eventloop-thread-1) gRPC Server started on 0.0.0.0:9000 [SSL enabled: false]
2021-11-08 14:26:58,647 INFO  [io.quarkus] (main) grpc-impl 1.0.0-SNAPSHOT native (powered by Quarkus 2.4.0.Final) started in 0.012s.
2021-11-08 14:26:58,647 INFO  [io.quarkus] (main) Profile prod activated.
2021-11-08 14:26:58,647 INFO  [io.quarkus] (main) Installed features: [cdi, grpc-server, smallrye-context-propagation, vertx]
Waited 60seconds for target/quarkus.log to contain info about the listening port and protocol but no such info was found
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 61.958 s <<< FAILURE! - in com.redhat.quarkus.reproducer.service.impl.PlayerServiceImplIT
[ERROR] com.redhat.quarkus.reproducer.service.impl.PlayerServiceImplIT.testGet  Time elapsed: 0.042 s  <<< ERROR!
java.lang.RuntimeException: java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
Caused by: java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details
```

### Expected behavior

The IntegrationTests fail to start because the process status cannot be retrieved

### Actual behavior

The IntegrationTests start normally

### How to Reproduce?

The following repository contains a reproducer where the last commit contains the addition of the quarkus-vertx-http:

https://github.com/ruromero/grpc-deps-reproducer/tree/main/grpc-impl

### Output of `uname -a` or `ver`

Linux 5.14.14-200.fc34.x86_64 #1 SMP Wed Oct 20 16:15:12 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk 11.0.13 2021-10-19 OpenJDK Runtime Environment 18.9 (build 11.0.13+8) OpenJDK 64-Bit Server VM 18.9 (build 11.0.13+8, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.4.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)

### Additional information

_No response_

--- a/extensions/grpc/codegen/pom.xml
+++ b/extensions/grpc/codegen/pom.xml
@@ -111,6 +111,12 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-grpc-protoc-plugin</artifactId>
             <classifier>shaded</classifier>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
     </dependencies>
 

--- a/test-framework/common/src/main/java/io/quarkus/test/common/LauncherUtil.java
+++ b/test-framework/common/src/main/java/io/quarkus/test/common/LauncherUtil.java
@@ -29,6 +29,8 @@ import io.smallrye.config.SmallRyeConfig;
 
 public final class LauncherUtil {
 
+    public static final int LOG_CHECK_INTERVAL = 500;
+
     private LauncherUtil() {
     }
 
@@ -106,7 +108,7 @@ public final class LauncherUtil {
         int i = 0;
         while (i++ < 10) {
             try {
-                Thread.sleep(500);
+                Thread.sleep(LOG_CHECK_INTERVAL);
             } catch (InterruptedException ignored) {
 
             }
@@ -177,11 +179,13 @@ public final class LauncherUtil {
      * process is listening
      */
     static void updateConfigForPort(Integer effectivePort) {
-        System.setProperty("quarkus.http.port", effectivePort.toString()); //set the port as a system property in order to have it applied to Config
-        System.setProperty("quarkus.http.test-port", effectivePort.toString()); // needed for RestAssuredManager
-        installAndGetSomeConfig(); // reinitialize the configuration to make sure the actual port is used
-        System.clearProperty("test.url"); // make sure the old value does not interfere with setting the new one
-        System.setProperty("test.url", TestHTTPResourceManager.getUri());
+        if (effectivePort != null) {
+            System.setProperty("quarkus.http.port", effectivePort.toString()); //set the port as a system property in order to have it applied to Config
+            System.setProperty("quarkus.http.test-port", effectivePort.toString()); // needed for RestAssuredManager
+            installAndGetSomeConfig(); // reinitialize the configuration to make sure the actual port is used
+            System.clearProperty("test.url"); // make sure the old value does not interfere with setting the new one
+            System.setProperty("test.url", TestHTTPResourceManager.getUri());
+        }
     }
 
     /**
@@ -195,6 +199,7 @@ public final class LauncherUtil {
         private final CountDownLatch signal;
         private final AtomicReference<ListeningAddress> resultReference;
         private final Pattern listeningRegex = Pattern.compile("Listening on:\\s+(https?)://\\S*:(\\d+)");
+        private final Pattern startedRegex = Pattern.compile(".*Quarkus .* started in \\d+.*s.*");
 
         public CaptureListeningDataReader(Path processOutput, Duration waitTime, CountDownLatch signal,
                 AtomicReference<ListeningAddress> resultReference) {
@@ -213,9 +218,19 @@ public final class LauncherUtil {
 
             long bailoutTime = System.currentTimeMillis() + waitTime.toMillis();
             try (BufferedReader reader = new BufferedReader(new FileReader(processOutput.toFile()))) {
+                long timeStarted = Long.MAX_VALUE;
+                boolean started = false;
+                // generally, we want to start as soon as info about Quarkus having started is printed
+                // but just in case the line with http host and port is printed later, let's wait a bit more
                 while (true) {
                     if (reader.ready()) { // avoid blocking as the input is a file which continually gets more data added
                         String line = reader.readLine();
+
+                        if (startedRegex.matcher(line).matches()) {
+                            timeStarted = System.currentTimeMillis();
+                            started = true;
+                        }
+
                         Matcher regexMatcher = listeningRegex.matcher(line);
                         if (regexMatcher.find()) {
                             dataDetermined(regexMatcher.group(1), Integer.valueOf(regexMatcher.group(2)));
@@ -228,17 +243,26 @@ public final class LauncherUtil {
                         }
                     } else {
                         //wait until there is more of the file for us to read
-                        try {
-                            Thread.sleep(500);
-                        } catch (InterruptedException e) {
-                            unableToDetermineData(
-                                    "Thread interrupted while waiting for more data to become available in proccess output file: "
-                                            + processOutput.toAbsolutePath().toString());
+
+                        long now = System.currentTimeMillis();
+                        // if we have seen info that the app is started in the log a while ago
+                        // or waiting the the next check interval will exceed the bailout time, it's time to finish waiting:
+                        if (now + LOG_CHECK_INTERVAL > bailoutTime || now - 2 * LOG_CHECK_INTERVAL > timeStarted) {
+                            if (started) {
+                                dataDetermined(null, null); // no http, all is null
+                            } else {
+                                unableToDetermineData("Waited " + waitTime.getSeconds() + " seconds for " + processOutput
+                                        + " to contain info about the listening port and protocol but no such info was found");
+                            }
                             return;
                         }
-                        if (System.currentTimeMillis() > bailoutTime) {
-                            unableToDetermineData("Waited " + waitTime.getSeconds() + "seconds for " + processOutput
-                                    + " to contain info about the listening port and protocol but no such info was found");
+
+                        try {
+                            Thread.sleep(LOG_CHECK_INTERVAL);
+                        } catch (InterruptedException e) {
+                            unableToDetermineData(
+                                    "Thread interrupted while waiting for more data to become available in process output file: "
+                                            + processOutput.toAbsolutePath());
                             return;
                         }
                     }

